src/settings.py

```diff
@@ -161,222 +71,122 @@ def settings_set(key, value):
-            "global_settings_key_doesnt_exists", settings_key=key
-        )
 
-    settings[key]["value"] = settings[key]["default"]
-    _save_settings(settings)
+    settings = SettingsConfigPanel()
+    key = translate_legacy_settings_to_configpanel_settings(key)
+    return settings.reset(key, operation_logger=operation_logger)
 
 
-def settings_reset_all():
+@is_unit_operation()
+def settings_reset_all(operation_logger):
     """
     Reset all settings to their default value
 
     Keyword argument:
         yes -- Yes I'm sure I want to do that
 
     """
-    settings = _get_settings()
+    settings = SettingsConfigPanel()
+    return settings.reset(operation_logger=operation_logger)
 
-    # For now on, we backup the previous settings in case of but we don't have
-    # any mecanism to take advantage of those backups. It could be a nice
-    # addition but we'll see if this is a common need.
-    # Another solution would be to use etckeeper and integrate those
-    # modification inside of it and take advantage of its git history
-    old_settings_backup_path = (
-        SETTINGS_PATH_OTHER_LOCATION % datetime.utcnow().strftime("%F_%X")
-    )
-    _save_settings(settings, location=old_settings_backup_path)
 
-    for value in settings.values():
-        value["value"] = value["default"]
+class SettingsConfigPanel(ConfigPanel):
+    entity_type = "settings"
+    save_path_tpl = SETTINGS_PATH
+    save_mode = "diff"
 
-    _save_settings(settings)
+    def __init__(
+        self, config_path=None, save_path=None, creation=False
+    ):
+        super().__init__("settings")
 
-    return {
-        "old_settings_backup_path": old_settings_backup_path,
-        "message": m18n.n(
-            "global_settings_reset_success", path=old_settings_backup_path
-        ),
-    }
+    def _apply(self):
+        super()._apply()
 
+        settings = { k: v for k, v in self.future_values.items() if self.values.get(k) != v }
+        for setting_name, value in settings.items():
+            try:
+                trigger_post_change_hook(setting_name, self.values.get(setting_name), value)
+            except Exception as e:
+                logger.error(f"Post-change hook for setting failed : {e}")
+                raise
 
-def _get_setting_description(key):
-    return m18n.n(f"global_settings_setting_{key}".replace(".", "_"))
+    def get(self, key="", mode="classic"):
+        result = super().get(key=key, mode=mode)
 
+        if mode == "full":
+            for panel, section, option in self._iterate():
+                if m18n.key_exists(self.config["i18n"] + "_" + option["id"] + "_help"):
+                    option["help"] = m18n.n(self.config["i18n"] + "_" + option["id"] + "_help")
+            return self.config
 
-def _get_settings():
+        # Dirty hack to let settings_get() to work from a python script
+        if isinstance(result, str) and result in BOOLEANS:
+            result = BOOLEANS[result]
```

Comment:
```suggestion
        # Dirty hack to let settings_get() to work from a python script
        if isinstance(result, str) and result in ["True", "False"]:
            result = bool(result)
```

Sounds easier to read that way ?